test/Microsoft.AspNetCore.Server.KestrelTests/HttpsConnectionFilterTests.cs

```diff
@@ -330,6 +331,32 @@ public void ThrowsWhenNoServerCertificateIsProvided()
             }
         }
 
+        [Fact]
+        public async Task ClientHandshakeFailureLoggedAsInformation()
+        {
+            var logger = new HandshakerErrorLogger();
+
+            var serviceContext = new TestServiceContext(new HttpsConnectionFilter(
+                new HttpsConnectionFilterOptions
+                { ServerCertificate = _x509Certificate2 },
+                new NoOpConnectionFilter(),
+                logger)
+            );
+
+            using (var server = new TestServer(App, serviceContext, _serverAddress))
+            {
+                using (await HttpClientSlim.GetSocket(new Uri($"https://localhost:{server.Port}/")))
+                {
+                    // Close socket immediately
+                }
+
+                await logger.LogTcs.Task;
+
+                Assert.Equal(1, logger.LastEventId.Id);
+                Assert.Equal(LogLevel.Information, logger.LastLogLevel);
```

Comment:
Do you think `HttpsConnectionFilter` should take an `ILoggerFactory` instead of an `ILogger`, so it picks the category?